panel/models/quill.ts

```diff
@@ -101,13 +106,17 @@ export class QuillInputView extends HTMLBoxView {
       theme: theme
     });
 
-    // Hack Quill and replace document.getSelection with shadow.getSelection
-    // see https://stackoverflow.com/questions/67914657/quill-editor-inside-shadow-dom/67944380#67944380
-    this.quill.selection.getNativeRange = () => {
-      const selection = (this.shadow_el as any).getSelection();
-      const range = normalizeNative(selection);
-      return range;
-    };
+    // Apply only with getSelection() is defined (e.g. undefined on Firefox)
+    if (typeof this.quill.root.getRootNode().getSelection !== 'undefined') {
```

Comment:
Shouldn't this be inside the function? Or is `getSelection` never defined in FireFox?

Reply:
Not sure about the correct language to use there, `getSelection` isn't "supported" in the shadow DOM on Firefox, see https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot#instance_methods.

I think that then when `document.getSelection()` is called on Firefox, it "pierces" through the shadow DOM. Of course, I have no idea what I'm talking about.

Reply:
>I have no idea what I'm talking about.

Fake til you make it 😄 That certainly sounded convincing.